openedx/adg/admin_dashboard/applications/views.py

```diff
@@ -57,3 +66,112 @@ def get_context_data(self, user):
+
+        self.is_user_adg_admin = is_adg_admin(request.user)
+        if self.is_user_adg_admin:
+            self.business_line_titles = list(BusinessLine.objects.titles())
+
+        self.filter_by_business_line = request.GET.get('filter-by', DEFAULT_BUSINESS_LINE_FILTER)
+        self.business_line_query = request.GET.getlist('business-line', self.business_line_titles)
+
+    def get_queryset(self):
+        """
+        Get queryset based on the type of application listing page
+        """
+        applications = UserApplication.objects.submitted_applications().filter(
+            status=self.application_type
+        ).select_related('user', 'user__profile', 'user__application_hub')
+
+        self.total_applications = applications.count()
+
+        filtered_applications = get_filtered_applications(
+            applications,
+            self.query,
+            self.omni_courses_query_values,
+            self.bu_courses_query_values,
+            self.filter_by_business_line,
+            self.business_line_query,
+            self.is_user_adg_admin
+        )
+
+        filtered_applications = filtered_applications.order_by(*self.get_ordering())
+        return filtered_applications
+
+    def get_paginate_by(self, queryset):
+        """
+        Allows dynamic pagination as selected by the user
+        """
+        self.paginate_by = self.request.GET.get('paginate-by', self.paginate_by)
+        return self.paginate_by
+
+    def get_context_data(self, *, object_list=None, **kwargs):
+        """
+        Prepare and return the extra context required by the listing pages i.e query params,
+        application type, total number of applications etc
+        """
+        context = super().get_context_data(object_list=object_list, **kwargs)
+
+        business_line_options = None
+        if self.is_user_adg_admin:
+            business_line_options = [DEFAULT_BUSINESS_LINE_FILTER, *self.business_line_titles]
+
+        context.update({
+            'application_type': self.application_type,
+            'total_applications': self.total_applications,
+            'business_lines': business_line_options,
+            'filter_by_business_line': self.filter_by_business_line,
+            'query': self.query,
+            'omni_courses_statuses': self.omni_courses_query_values,
+            'bu_courses_statuses': self.bu_courses_query_values,
+            'business_line_query': self.business_line_query,
+            'paginate_by': self.paginate_by,
+        })
+
+        return context
+
+    def get_ordering(self):
+        """
+        Get the fields to order the applications queryset
+        """
+        date_to_sort_applications = '-modified' if self.application_type == UserApplication.ACCEPTED \
+            else '-user__application_hub__submission_date'
+
+        return (date_to_sort_applications, '-user__application_hub__program_prereq_courses_status',
+                '-user__application_hub__bu_prereq_courses_status')
```

Comment:
What can we do about this one? We need ordering for the queryset of application review.

Reply:
Created a helper function for ordering as well. You can simply use that in the review page